optaplanner-wb-screens/optaplanner-wb-guided-rule-editor/optaplanner-wb-guided-rule-editor-backend/src/main/java/org/optaplanner/workbench/screens/guidedrule/backend/server/service/ScoreHolderServiceImpl.java

```diff
@@ -0,0 +1,134 @@
+    public ScoreHolderServiceImpl( final KieProjectService kieProjectService,
+                                   @Named("ioStrategy") final IOService ioService,
+                                   final GlobalsEditorService globalsEditorService,
+                                   final DataModelerService dataModelerService ) {
+        this.kieProjectService = kieProjectService;
+        this.ioService = ioService;
+        this.globalsEditorService = globalsEditorService;
+        this.dataModelerService = dataModelerService;
+    }
+
+    public Collection<String> extractProjectScoreTypeFqns( final Path projectPath ) {
+        List<Path> classUsages = dataModelerService.findClassUsages( projectPath,
+                                                                     PlanningSolution.class.getName() );
+
+        Set<String> result = new HashSet<>();
+        for ( Path path : classUsages ) {
+            result.addAll( extractSolutionScoreTypeFqns( path ) );
+        }
+        return result;
+    }
+
+    private Collection<String> extractSolutionScoreTypeFqns( final Path solutionObjectPath ) {
+        org.uberfire.java.nio.file.Path source = Paths.convert( kieProjectService.resolvePackage( solutionObjectPath ).getPackageMainResourcesPath() );
+        org.uberfire.java.nio.file.Path sourcePackage = Files.isDirectory( source ) ? source : source.getParent();
+
+        String sourceSolutionFileName = solutionObjectPath.getFileName().substring( 0,
+                                                                                    solutionObjectPath.getFileName().indexOf( "." ) );
+        org.uberfire.java.nio.file.Path sourceScoreHolderGlobalPath = sourcePackage.resolve( sourceSolutionFileName + SCORE_HOLDER_GLOBAL_FILE_SUFFIX );
+
+        boolean scoreHolderGlobalFileExists = ioService.exists( sourceScoreHolderGlobalPath );
+
+        if ( scoreHolderGlobalFileExists ) {
+            GlobalsModel globalsModel = globalsEditorService.load( Paths.convert( sourceScoreHolderGlobalPath ) );
+            return globalsModel.getGlobals().stream()
+                    .filter( global -> "scoreHolder".equals( global.getAlias() ) )
+                    .map( Global::getClassName )
+                    .collect( Collectors.toList() );
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public Collection<BendableScoreLevelsWrapper> getBendableScoreLevelsSize( final Path projectPath ) {
+        List<Path> classUsages = dataModelerService.findClassUsages( projectPath,
+                                                                     PlanningSolution.class.getName() );
+
+        List<BendableScoreLevelsWrapper> resultList = new ArrayList<>( classUsages.size() );
+
+        for ( Path solutionPath : classUsages ) {
+            String solutionString = ioService.readAllString( Paths.convert( solutionPath ) );
+
+            GenerationResult generationResult = dataModelerService.loadDataObject( solutionPath,
```

Comment:
You might like to check ```DataModelerService.loadDataObject(..)``` can handle the ```solutionPath``` AFAIK ```DataModelerService.findClassUsages(..)``` returns paths for any type of file referencing ```PlanningSolution```; so it could include Guided Rules or Guided Decision Tables etc.